### DISCUSSION TITLE
Using FastAPI with Next.js and NextAuth

### DISCUSSION THREAD

**Discussion by TCatshoek** [nextauthjs/next-auth#8064] [Ideas]
Hi everyone, I wrote a little FastAPI library to make it easy to use NextAuth generated tokens in FastAPI: https://github.com/TCatshoek/fastapi-nextauth-jwt. It also does NextAuth compatible CSRF protection.

I saw some previous discussions in here about this, but none of them matched exactly and they were from a while ago so I figured I'd start a dicussion here. 

I'm still working on some examples too, but I'd love to hear what you think!

**Comment by johntopdev0011** [comment]
Thank you very much.
as you mentioned. there are many discussing with next.js and fastapi.
But there is no matched with them.

I am trying to auth with them, but I am not going to try because of my broken skill.
Please help me!
Thank you!

**Reply by TCatshoek** [reply]
I added a very basic usage example here: https://github.com/TCatshoek/fastapi-nextauth-jwt/tree/014afc078614ab135d182df33e78d742a26cfa28/examples/simple/simple

I hope that helps!

**Comment by johntopdev0011** [comment]
Thank you very much

**Comment by ErikvdVen** [comment]
I am completely new to NextJS so forgive me if this a dumb question, but doesn't this make your fastapi endpoints really depend on the NextJS NextAuth payload? And thus too coupled?

I just started building a NextJS front-end for my FastAPI backend so I am doing some research on this topic, but doesn't Auth.JS already provide solutions on this topic? https://authjs.dev/guides/integrating-third-party-backends

I would say I want NextAuth to authenticate against my fastapi auth endpoint and store the returned accesstoken on the NextJS side (encrypted preferably but I guess that's already the idea) and use that token attached to a header, to authenticate for all my other requests.

**Reply by TCatshoek** [reply]
Not a dumb question at all! You're completely right about the fact that doing it this way ties your fastapi endpoints to the way nextauth creates the JWTs, since all it does is just verify and decrypt the JWT created by nextauth.

This was perfect for my simple use case of having nextjs and fastapi behind a reverse proxy, with the client making requests to fastapi directly, and allowed me to sort of treat it as writing endpoints for the nextjs application itself, but in python. 

If this coupling is a dealbreaker or if your use case is more advanced, I suppose doing it like described in the link you posted is more suitable.

**Reply by ErikvdVen** [reply]
Thank you so much for your detailed and quick answer! Makes completely sense. In that case, I will try to first get things working without your library, but I'll bookmark it anyway as a backup :)

**Reply by TCatshoek** [reply]
No problem, happy to help!
